### ISSUE TITLE
Alerting: (email) image never updates on subsequent notification policy repeat interval and is missing the red "ALERTING" annotation

### ISSUE THREAD

**Issue by salavessa** [grafana/grafana#75112]
### What happened?

In a notification policy with "repeat interval" configured, the "repeated" email notifications send the up-to-date "text values" but the attached image is always the same.
Also, from all the tests I've made. the red "ALERTING" annotation (vertical line) which makes the alert trigger is never present in the image - maybe because by the time grafana image renderer fetches the "solo" page the annotation hasn't been created yet?

### What did you expect to happen?

I would expect that the image sent as an attachment in the email to be updated to the "current" selected panel state and that the "ALERTING" annotation relative to the alert being fired to be visible in the image.
E.g. if I have a condition running every 5 minutes, failing for 2 hours, with a notification policy repeat interval of `5m` then I get around 24 emails with the exact same image attached but up-to-date failing "values".

### Did this work before?

This functionality was fine in the legacy alerting mechanism: the image was up to date by the time of each "repeated" email and the annotation relative to the alert being fired was present from the first email alert.
Not sure if it ever worked in the unified alerting.

### How do we reproduce it?

1. create an email notification policy with repeat interval
2. create an alert rule pointing at a dashboard+panel
3. make the alert fire all the time
4. check the "ALERTING" annotation is not present in email (but present in the panel)
5. check that there is the same image for each "repeated" email notifications

### Is the bug inside a dashboard panel?

No.

### Environment (with versions)?

Grafana: 10.1.2
OS: linux
Browser: Chrome

### Grafana platform?

Docker

**Comment by usmangt** [comment]
Hi @salavessa,

Thanks for opening this issue.

I will forward this to our Engineers so that they can have a look and might be able to tell more details about it.

**Comment by armandgrillet** [comment]
Repeats of an existing notification will get the same annotation thus the same image. We would have to update the annotation. We do not want the ruler to take a screenshot at every eval cycle but others options are possible which we need to investigate them.

"Also, from all the tests I've made. the red "ALERTING" annotation (vertical line) which makes the alert trigger is never present in the image - maybe because by the time grafana image renderer fetches the "solo" page the annotation hasn't been created yet?" is a separate bug also called annotations, could you please create a separate issue for it?

**Comment by salavessa** [comment]
> Repeats of an existing notification will get the same annotation thus the same image. We would have to update the annotation. We do not want the ruler to take a screenshot at every eval cycle but others options are possible which we need to investigate them.

For our scenarios it makes it really difficult to quickly identify the trend/progress of a breaching metric if we get the initial image only.
All we are looking for is the exact same behaviour of the legacy alerting "repeated" email notifications.
Maybe this is not the best thing for all users/systems/panels/metrics so a switch to enable/disable this behaviour would be great (e.g. a boolean selector in the notification policy) as we are aiming to take/receive an up-to-date screenshot of the specific panel at every eval cycle on a firing state, but we don't really want for the annotation to be "updated" or new ones created for each "repeated" email notifications.

> "Also, from all the tests I've made. the red "ALERTING" annotation (vertical line) which makes the alert trigger is never present in the image - maybe because by the time grafana image renderer fetches the "solo" page the annotation hasn't been created yet?" is a separate bug also called annotations, could you please create a separate issue for it?

Issue https://github.com/grafana/grafana/issues/75491 created as requested.

Thanks

**Comment by armandgrillet** [comment]
Thank you for the additional issue, your scenarios make sense 👍.

**Comment by apollo13** [comment]
We would also love to see the image get updated, seeing a trend evolve from the images is way easier than from the values alone.

**Comment by grobinson-grafana** [comment]
Hi! :wave: At present screenshots are not re-taken over the lifetime of an alert, just when an alert fires and resolves. The reason for this is that taking screenshots is very computationally expensive, and taking a new screenshot on each evaluation would add too much load to the image renderer.

We could consider setting a maximum age of screenshots (i.e. 10 minutes), and if the alert is still firing after 10 minutes the screenshot is re-taken. Would something like that work?

**Comment by salavessa** [comment]
Hi @grobinson-grafana,

That would definitely work and, as I see it, it's better by a million miles than the current behavior where we can have an alert being fired for many hours and have no visibility of the trend.
Ideally the screenshot maximum age should be configurable.

Another thing I noticed and could probably be easily and massively improved is that, when there's more than one alert "dimension" failing (or alerts that target the same dashboard+panel) we get many screenshots of the same panel over an over in the same email, it would be good if that could be aggregated somehow as the size of the email gets huge and it's necessary to perform a tedious/endless scroll to look into all email alerts.

Thanks!

**Comment by apollo13** [comment]
@grobinson-grafana Yes 10 minutes would be more than enough for us. Even every half our would be an improvement. If it is not to hard to support a global (per alert would probably be overkill) setting to configure the maximum age would be nice as well (for people willing to take the perf hit).

**Comment by MonicaMagoniCom** [comment]
Any news on this?

**Comment by apollo13** [comment]
Unstale
